### DISCUSSION TITLE
I wrote a description of how to modify Presets

### DISCUSSION THREAD

**Discussion by Hypersonic-Popcorn** [lllyasviel/Fooocus#2429] [Show and tell]
If you want to modify a preset to load different checkpoints or change the number of steps but don't know how, [this file](https://github.com/ForgetfulWasAria/docs/blob/main/preset.md) will help. 

Honestly, it's overkill for most things. You can simply open up the preset file and make changes. It's quite long, around 2k words as I tried to explain everything for the non-tech savy among us. Also, there are some settings, marked with ????, that I didn't know how to describe. If this is useful, feel free to make additions, corrections, and also feel free to host it where ever is most useful. Thanks!

**Comment by mashb1t** [comment]
Thanks a lot, it's pretty extensive and may help understand people who can't read the python code how to configure Fooocus.

The latest version of Fooocus ([2.2.0](https://github.com/lllyasviel/Fooocus/releases/tag/2.2.0)) introduced additional config options, which you might want to add.

- default_loras_min_weight
- default_loras_max_weight
- default_max_lora_number
- default_save_metadata_to_images
- default_metadata_scheme
- metadata_created_by

I think all of them are self-explanatory.

**Reply by Hypersonic-Popcorn** [reply]
Done. I also added hand-wavy descriptions for default_inpaint_mask_model, default_inpaint_mask_cloth_category, and default_inpaint_mask_sam_model. Essentially, put your model here. I have no experience with these model types.

Also, I assume "default_black_out_nsfw" is a boolean value that controls whether nsfw is shown meant for those who expose Fooocus to the public.

**Reply by mashb1t** [reply]
Great!
"default_black_out_nsfw" is part of my fork and PR https://github.com/lllyasviel/Fooocus/pull/958. No need to add it for now.
Please make sure you've not accidentally crawled my fork instead of Fooocus.

**Reply by Hypersonic-Popcorn** [reply]
Ok, I see. I've been using your fork myself. Originally, I had a section for your fork with the additional options you support but since all of them except black_out_nsfw are now in the main branch I mashed them together. In any case, I removed default_black_out_nsfw for now and I see where it is in the commit.
